anesthetic/read/cobaya.py

```diff
@@ -0,0 +1,80 @@
+"""Read MCMCSamples from cobaya chains."""
+import os
+import glob
+import numpy as np
+from anesthetic.read.utils import remove_burn_in
+from anesthetic.samples import MCMCSamples
+
+
+def read_paramnames(root):
+    r"""Read header of <root>.1.txt to infer the paramnames.
+
+    This is the data file of the first chain. It should have as many
+    columns as there are parameters (sampled and derived) plus an
+    additional two corresponding to the weights (first column) and the
+    logposterior (second column). The first line should start with a # and
+    should list the parameter names corresponding to the columns. This
+    will be used as label in the pandas array.
+    """
+    with open(root + ".1.txt") as f:
+        header = f.readline()[1:]
+        paramnames = header.split()[2:]
```

Comment:
Only split off the weights and likelihood values. Don't split off the last columns. For the example data here that might be tempting, but these columns will actually be different for runs with multiple likelihoods.

Reply:
Agreed.